src/s2_fos/interface.py

```diff
@@ -0,0 +1,266 @@
+        {'micro': 0.7919858573954036, 'macro': 0.7773981933829964, 'weighted': 0.8000682133515643})
+        micro: 0.85443
+        macro: 0.83920
+        weighted: 0.87635
+
+        For abstract missing
+        Best micro-f1 score: ({'thr_1': 0.5172413793103449, 'thr_2': 0.6206896551724138, 'thr_3': 0.6206896551724138},
+        {'micro': 0.7784503631961258, 'macro': 0.7374930547135901, 'weighted': 0.7847926929179073})
+        Average precision:
+        micro: 0.84874
+        macro: 0.81286
+        weighted: 0.86451
+        Args:
+            threshold_list_np ():
+            abstract_np ():
+
+        Returns:
+
+        """
+        argmax_idx = np.argpartition(-threshold_list_np, kth=5, axis=1)[:, :5]
+        assigned_labels = np.zeros(threshold_list_np.shape)
+        for row_idx, row in enumerate(threshold_list_np):
+            for idx_n, idx in enumerate(argmax_idx[row_idx]):
+                abstract = abstract_np[row_idx]
+                if abstract is None or abstract == "":
+                    if self._config.thr_1_no_abstract is not None or self._config.thr_2_no_abstract is not None:
+                        thr_1, thr_2, thr_3 = (self._config.thr_1_no_abstract, self._config.thr_2_no_abstract,
+                                               self._config.thr_3_no_abstract)
+                    else:
+                        thr_1, thr_2, thr_3 = (thr_1_no_abstract, thr_2_no_abstract, thr_3_no_abstract)
+                else:
+                    if self._config.thr_1 is not None or self._config.thr_2 is not None:
+                        thr_1, thr_2, thr_3 = self._config.thr_1, self._config.thr_2, self._config.thr_3
+                    else:
+                        thr_1, thr_2, thr_3 = thr_1_w_abstract, thr_2_w_abstract, thr_3_w_abstract
+
+                if row[idx] >= thr_1 and idx_n == 0:
+                    assigned_labels[row_idx, idx] = 1
+                elif row[idx] >= thr_2 and idx_n >= 1:
+                    assigned_labels[row_idx, idx] = 1
+                elif row[idx] >= thr_3 and idx_n >= 2:
+                    assigned_labels[row_idx, idx] = 1
+                else:
+                    assigned_labels[row_idx, idx] = 0
+        return assigned_labels
+
+    def predict_labels_from_np(self, data_np: np.array) -> np.array:
+        """
+        Predicts probabilities of the FoS
+        Args:
+            data_np (np.array): numpy array of the title, abstract, journal_names
+
+        Returns: numpy array of probability scores for each of the fields of study
+
+        """
+        text = [f'{self.tokenizer.sep_token}'.join([(example[i] if example[i] is not None else '')
+                                                    for i in range(3)]) for example in data_np]
+        inputs = self.tokenizer(
+            text,
+            return_tensors="pt",
+            padding=True,
+            truncation=True,
+            max_length=128,
```

Comment:
This 128 seems low - is this how you trained? Cutting off?

Reply:
Yes I was cutting it off, agree 128 seems low